x/concentrated-liquidity/README.md

```diff
@@ -1045,6 +1045,90 @@ feeChargeTotal = amountIn.Mul(swapFee)
 
 TODO
 
+#### TWAP Integration
+
+In the context of twap, concentrated liquidity pools function differently from
+CFMM pools.
+
+There are 2 major differences that stem from how the liquidity is added and removed in concentrated-liquidity.
+
+The first one is given by the fact that a user does not provide liquidity at pool creation time. Instead,
+they have to issue a separate message post-pool creation. As a result, there can be a time where there
+is no valid spot price initialized for a concentrated liquidity pool. When a concentrated liquidity
+pool is created, the `x/twap` module still initializes the twap records. However, these records are invalidated
+by setting the "last error time" field to the block time at pool creation. Only adding liquidity to the pool
+will initialize the spot price and twap records correctly. One technical detail to note is that adding
+liquidity in the same block as pool creation will still set the "last error time" field to the block time
+despite spot price already being initialized.
```

Comment:
Can we create an issue for this and add it here in this thread?

Reply:
Discussed offline - fine to keep as is but expand on why this is working this way

Reply:
Expanded here: [d1c5d46](https://github.com/osmosis-labs/osmosis/pull/4831/commits/d1c5d461e9c3720a83022d1f49cffd3d55e9454e)